artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -617,4 +650,55 @@ public static void routeMirrorCommand(ActiveMQServer server, Message message, Tr
       server.getPostOffice().route(message, ctx, false);
    }
 
+   class PagedRouteContext implements RouteContextList {
+
+      private final List<Queue> listRepresentation;
+      private final List<Queue> emptyList = Collections.emptyList();
+
+      PagedRouteContext() {
+         listRepresentation = new ArrayList<>(1);
+         listRepresentation.add(snfQueue);
```

Comment:
If the snfQueue durability is essentially fixed, cant we pre-determine the answers/return values and simplify much of the method code?